pkg/lib/file_header.go

```diff
@@ -43,62 +40,144 @@ type FileHeader struct {
 	Receiver             string
 	FileCreatedDate      string
 	FileCreatedTime      string
-	FileIdNumber         int64
-	PhysicalRecordLength int64
-	BlockSize            int64
+	FileIdNumber         string
+	PhysicalRecordLength int64 `json:",omitempty"`
+	BlockSize            int64 `json:",omitempty"`
 	VersionNumber        int64
 }
 
 func (h *FileHeader) Validate() error {
 	if h.RecordCode != "01" {
-		return fmt.Errorf("FileHeader: invalid record code")
+		return fmt.Errorf(fmt.Sprintf(fhValidateErrorFmt, "RecordCode"))
 	}
-	if h.Sender != "0004" {
-		return fmt.Errorf("FileHeader: invalid sender")
+	if h.Sender == "" {
+		return fmt.Errorf(fmt.Sprintf(fhValidateErrorFmt, "Sender"))
 	}
-	if h.PhysicalRecordLength != 80 {
-		return fmt.Errorf("FileHeader: invalid physical record length")
+	if h.Receiver == "" {
+		return fmt.Errorf(fmt.Sprintf(fhValidateErrorFmt, "Receiver"))
 	}
-	if h.BlockSize != 1 {
-		return fmt.Errorf("FileHeader: invalid block size")
+	if h.FileCreatedDate == "" {
+		return fmt.Errorf(fmt.Sprintf(fhValidateErrorFmt, "FileCreatedDate"))
+	} else if !util.ValidateData(h.FileCreatedDate) {
+		return fmt.Errorf(fmt.Sprintf(fhValidateErrorFmt, "FileCreatedDate"))
+	}
+	if h.FileCreatedTime == "" {
+		return fmt.Errorf(fmt.Sprintf(fhValidateErrorFmt, "FileCreatedTime"))
+	} else if !util.ValidateTime(h.FileCreatedTime) {
+		return fmt.Errorf(fmt.Sprintf(fhValidateErrorFmt, "FileCreatedTime"))
+	}
+	if h.FileIdNumber == "" {
+		return fmt.Errorf(fmt.Sprintf(fhValidateErrorFmt, "FileIdNumber"))
 	}
 	if h.VersionNumber != 2 {
-		return fmt.Errorf("FileHeader: invalid version number")
+		return fmt.Errorf(fmt.Sprintf(fhValidateErrorFmt, "VersionNumber"))
 	}
 
 	return nil
 }
 
-func (h *FileHeader) Parse(line string) error {
-	if n := utf8.RuneCountInString(line); n < fileHeaderLength {
-		return fmt.Errorf("FileHeader: length %d is too short", n)
+func (h *FileHeader) Parse(data string) (int, error) {
+
+	var line string
+	var err error
+	var size, read int
+
+	if length := util.GetSize(data); length < 2 {
+		return 0, fmt.Errorf(fmt.Sprintf(fhParseErrorFmt, "record"))
+	} else {
+		line = data[:length]
 	}
 
-	h.RecordCode, _ = util.EntryParser(line[0:3], ",")
-	h.Sender, _ = util.EntryParser(line[3:8], ",")
-	h.Receiver, _ = util.EntryParser(line[8:14], ",")
-	h.FileCreatedDate, _ = util.EntryParser(line[14:21], ",")
-	h.FileCreatedTime, _ = util.EntryParser(line[21:26], ",")
-	h.FileIdNumber, _ = util.EntryParserToInt(line[26:30], ",")
-	h.PhysicalRecordLength, _ = util.EntryParserToInt(line[30:33], ",")
-	h.BlockSize, _ = util.EntryParserToInt(line[33:35], ",")
-	h.VersionNumber, _ = util.EntryParserToInt(line[35:37], "/")
+	// RecordCode
+	if h.RecordCode, size, err = util.ReadField(line[read:]); err != nil {
```

Comment:
How are we validating the length of `line` will be long enough for each `util.ReadField(..)` call? I can see this pattern throwing panics on short inputs.